docs/SemanticARC.md

```diff
@@ -128,8 +145,41 @@ In order to simplify this, we will make the following changes:
    * @unowned @safe
    * @forwarding
 
+[EE] We really need a good definition of all these conventions here.
+I strongly believe that the local/global notation would help a lot for a good formal definition of the conventions.
+It's hard for the compiler to reason anything about global RCs. But local RCs are easily statically computable and for me, semantic ARC is all about dealing with local RCs.
+
+So we could define:
+
+@owned operand: local RC must be 1 and is decremented to 0 by the instruction, potentially destroying the object.
+@owned result: Returns the object with a local RC of 1
+@guaranteed operand: local RC must be 1 and is not modified by the instruction
+@unowned operand: local RC must be 0 and is not modified by the instruction.
+@unowned result: Returns the object with a local RC of 0
+@forwarding: Does not change the local RC which can be either 0 or 1
+
+Note that the local RC can only be 0 or 1. It doesn't make sense to have a local RC > 1.
+
+ad @unowned: what do you mean with @unsafe and @safe
+
+ad SILArguments: given the definitions above, what is @owned for a SILArgument? In other words, do we need @owned for SILArgument in addition to @guaranteed and @unowned?
+
+ad @guaranteed: There is a potential alternative definition of @guaranteed for function parameters: the local + global RC must be at least 1. In case the optimizer can figure out that a function does not bring down the global RC to 0, there is no need to have a local RC of 1 for a @guaranteed parameter. So maybe it's better to use a different name for the pure local +1RC convention.
+
+ad @forwarding: I think we don't need this in SIL syntax. It's the default if not @owned, @guaranteed, etc.
+
+
 @forwarding is a new convention that we add to reduce the amount of extra instructions needed to implement this scheme. @forwarding is a special convention intended for instructions that forward RC Identity that for simplictuy will be restricted to forwarding the convention of their def instruction to all of the uses of that instruction. Of course, for forwarding instructions with multiple inputs, we require that all of the inputs have the same convention.
 
+[EE] Do we really need this restriction? What about this example?
+
+bb0(%0 : @guaranteed %X, %1 : @owned %Y):
+  %s = struct $Foo(%0 : %X, %1 : Y)
+  // ... some code
+  %s1 = struct_extract %s, #x
+  %s2 = struct_extract %s, #y
+  %a = apply %f(%s1, %s2) : $@convention(thin) (@guaranteed X, @owned Y)
+
```

Comment:
Having the restriction that all SSA values need to have only one convention associated with them simplifies the model. Being able to mix like this just complicates the model in an unnecessary way.